File: ex111.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def cal_depth(self, root, depth, p):
        global minimum
        
        if root != None:
            #Record the root of subtree
            p = root
            
            self.cal_depth(root.left, depth + 1, p)
            self.cal_depth(root.right, depth + 1, p)
        else:
            if (depth < minimum) and (p.left == None) and (p.right == None):
                minimum = depth

    
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        
        global minimum, p
        
        minimum = 2**31
        
        #Two specific conditions with no node
        if root == None:
            return 0
        

        self.cal_depth(root, 0, root)   
        
        return minimum
        

File: code/ex111.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root == NULL)    return 0;
        queue<TreeNode*> que;
        que.push(root);
        int depth = 0;
        while (!que.empty()) {
            int size = que.size();
            for (int z = size; z > 0; z--) {
                TreeNode* out = que.front();    que.pop();
                if (out->left == NULL && out->right == NULL)    return depth + 1;
                if (out->left != NULL)    que.push(out->left);
                if (out->right != NULL)    que.push(out->right);
            }
            depth++;
        }
        return -1;
    }
};